### PR TITLE
resolves: 117 discoverer

### LINKED ISSUE #117: Discoverer
Add discoverer configuration: 
- Reframe's config `discoverer.py` containing available systems/environments.
- Working example of _config/machines/discoverer.json_ file

### LINKED ISSUE #175: Add a dry-run option
Add an option that allows to dry-run a benchmark and export the replaced configuration files in order to manually verify generated config. 

This would be extremely useful for making sure of configurations before executing on HPC resources.

### PR DESCRIPTION
**Pull request by JavierCladellas**
- Closes #117
- Closes #175 

Done: 
- [x] Added ReFrame config for discoverer
- [x] Added shell scripts for gaya and discoverer for runners to launch `execute-benchmark`
- [x] Improved the HpcSystem dispatcher
- [x] Added a dry run option to verify job scripts that will be submitted 
- [x] Updated the thermal_bridges case config
- [x] Updated the _benchmark.yml_ workflow

### CODE REVIEW

File: src/feelpp/benchmarking/reframe/config/machineConfigs/discoverer.py [hunk 1]
```diff
@@ -0,0 +1,50 @@
...
+                {
+                    'name': 'cn',
+                    'scheduler': 'slurm',
+                    'launcher': 'srun',
+                    'max_jobs': 8,
+                    'access': ['--partition=cn --account=ehpc-dev-2024d05-047 --qos=ehpc-dev-2024d05-047'],
+                    'environs': ['default'],
+                    'prepare_cmds': [
+                        'source /etc/profile.d/modules.sh',
+                        'export MODULEPATH=/opt/software/modulefiles',
```
Comment (vincentchabannes) [inline src/feelpp/benchmarking/reframe/config/machineConfigs/discoverer.py:18]: It is not linked to discoverer and should be removed, no?

File: src/feelpp/benchmarking/reframe/config/machineConfigs/discoverer.py [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+                        'export OMP_NUM_THREADS=1' # Just in case
```
Comment (vincentchabannes) [inline src/feelpp/benchmarking/reframe/config/machineConfigs/discoverer.py:19]: That should not be appear here, it's the application which can define this variables

Comment (JavierCladellas) [reply src/feelpp/benchmarking/reframe/config/machineConfigs/discoverer.py:19]: Should we have a benchmark/application configuration field like 
```json 
"env_variables":{ "OMP_NUM_THREADS":1 }
``` 
that will export environment variables at the start of the benchmark?

Comment (vincentchabannes) [reply src/feelpp/benchmarking/reframe/config/machineConfigs/discoverer.py:17]: yes

Comment (JavierCladellas) [reply src/feelpp/benchmarking/reframe/config/machineConfigs/discoverer.py]: Ok, I have added the `env_variables` field under the application config root. 
For containers, they should be passed using their native option under the `options` field. e.g for apptainer: 
```json
 "options": [
   "--env OMP_NUM_THREADS=1"
]
```

**Review by vincentchabannes — APPROVED** [review]

**Pull request merged**